synadm/cli/notice.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+# synadm
+# Copyright (C) 2022 Philip (a-0)
+#
+# synadm is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# synadm is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""server notice-related CLI commands
+"""
+
+import re
+import click
+
+from synadm import cli
+
+
+@cli.root.group()
+def notice():
+    """ Send server notices to local users
+    """
+
+
+@notice.command(name="send")
+@click.option(
+    "--from-file", "-f", default=False, show_default=True, is_flag=True,
+    help="""Interpret arguments as file paths instead of notice content and
+    read content from those files""")
+@click.option(
+    "--paginate", type=int, default=100, show_default=True, help="""Sets how
+    many users will be retrieved from the server at once""")
+@click.option(
+    "--to-regex", "-r", default=False, show_default=True, is_flag=True,
+    help="Interpret TO as regular expression")
+@click.option(
+    "--match-list-length", "-l", type=int, default=10, show_default=True,
+    help="""Length of the displayed list of matched receivers. Does not impact
+    sending behavior""")
```

Comment:
Also here I think there is just a little room for improvement. It should be made clear that this option is about the recipient list that is only shown in the dry run. It kind of tells that but it could be made even better. Maybe just adding some to the first sentence would do the trick:

```
Length of the displayed list of potential recipients shown in the confirmation prompt.
```

```
Length of the displayed list of matched recipients shown in the confirmation prompt.
```

```
Length of the displayed list of recipients shown in the confirmation prompt.
```

```
How many recipients should be shown in the confirmation prompt?
```

```
How many recipients should be shown in the message preview?
```

Just brainstorming here, might not all be perfect...

Reply:
Also something like this could be added for exactly documenting what this option is about and what it is not:

```
This option is ignored when the global `--batch` flag is set, since there won't be a confirmation prompt. 
```

Reply:
I included the second suggestion you made, it's a good improvement. However, I think adding this (explicitly stating it is about the confirmation prompt) makes it clear enough that it won't have any effect if you switch off the confirmation prompt. I would argue in favor of brievety here.